src/tools/crawling_tools.py

```
@@ -1038,3 +1354,110 @@ async def _update_sources(ctx: Context, source_content_map: Dict[str, str], sour
+    
+    # Clean and deduplicate
+    clean_apis = []
+    for api in mentioned_apis:
+        api_clean = api.strip('`@()')
+        if len(api_clean) > 2 and api_clean not in clean_apis:
+            clean_apis.append(api_clean)
+    
+    return clean_apis[:20]  # Return top 20
+
+
+def _extract_code_blocks(content: str) -> List[Dict[str, Any]]:
```

Ruff (0.12.2)

Use `list` instead of `List` for type annotation

Replace with `list`

(UP006)